app/src/main/java/org/eyeseetea/malariacare/domain/usecase/PullLoggedUserAttributesUseCase.java

```diff
@@ -0,0 +1,98 @@
+/*
+ * Copyright (c) 2017.
+ *
+ * This file is part of QA App.
+ *
+ *  Health Network QIS App is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  Health Network QIS App is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with Foobar.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package org.eyeseetea.malariacare.domain.usecase;
+
+import org.eyeseetea.malariacare.data.database.utils.PreferencesState;
+import org.eyeseetea.malariacare.domain.boundary.IUserAttributesRemoteDataSource;
+import org.eyeseetea.malariacare.domain.boundary.IUserLocalDataSource;
+import org.eyeseetea.malariacare.domain.boundary.executors.IAsyncExecutor;
+import org.eyeseetea.malariacare.domain.boundary.executors.IMainExecutor;
+import org.eyeseetea.malariacare.domain.entity.UserAccount;
+import org.eyeseetea.malariacare.domain.entity.UserAttributes;
+import org.eyeseetea.malariacare.domain.exception.PullUserAttributesException;
+
+public class PullLoggedUserAttributesUseCase implements UseCase {
+
+    public interface Callback {
+        void onSuccess(UserAttributes userAttributes);
+
+        void onError(UserAttributes userAttributes);
+    }
+
+    private final IAsyncExecutor mAsyncExecutor;
+    private final IMainExecutor mMainExecutor;
+    private final IUserLocalDataSource userLocalDataSource;
```

Comment:
A use case should be agnostic with data, It should execute UserAccountRepository and this to retrieve data from the local data source or remote data. The logic to execute remote or local can be implemented for several ways, one can be with time if the user data stored in local has more of 1 days for example then return from the remote. Sometimes this logic can be realized by a parameter to use case and repository like NetworkStrategy (NetworkFirst and LocalFirst). I think that Current UserRepositoy (with login and logout) should be rename to AuthenticationManager like in PictureApp and create new UserAccountRepository to retrieve UserAccount from local or remote according to needs